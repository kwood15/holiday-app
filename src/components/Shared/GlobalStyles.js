import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Open+Sans');

  * {
    box-sizing: border-box;
  }

  #root {
    display: flex;
    min-height: 100%;
    flex-direction: column;
  }

  html {
    height: 100%;
    margin: 0;
    padding: 0;
  }

  body {
    font-family: 'Open Sans', sans-serif;
    height: 100%;
    background-color: ${props => props.theme.bodyColor};
    font-size: 1rem;
    line-height: 1.2;
    margin: 0;
    padding: 0;
  }

  ul, ol {
    margin: 0;
    padding: 0;
  }

  li {
    list-style: none;
  }

  a {
    color: ${props => props.theme.primaryColor};
    text-decoration: none;
  }

  p {
    color: ${props => props.theme.textColor};
    font-size: 1rem;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
  }

  h1, h2, h3, h4, h5, h6 {
    color: ${props => props.theme.headingColor};
  }

  h1 {
    font-size: 1.5rem;
  }

  legend {
    font-weight: bold;
    color: ${props => props.theme.primaryColor};
  }

  fieldset {
    border: 0;
    padding: 0 0 1.5rem;
  }
`;
