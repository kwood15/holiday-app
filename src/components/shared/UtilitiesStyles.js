import { createGlobalStyle } from 'styled-components';

export const UtilitiesStyles = createGlobalStyle`
  *:focus {
    outline: 2px dotted ${props => props.theme.primaryColor};
  }

  .u-separator {
    color: ${props => props.theme.textColor};
    padding: 0 4px;
    font-size: 0.85em;
  }

  .u-lock-scroll {
    overflow: hidden !important;
  }

  .u-visible {
    display: block;
  }

  .u-hidden {
    display: none;
  }

  .u-margin-top-none {
    margin-top: 0;
  }

  .u-visually-hidden {
    position: absolute;
    overflow: hidden;
    clip: rect(0 0 0 0);
    height: 1px;
    width: 1px;
    margin: -1px;
    padding: 0;
    border: 0;
    top: 0;
    left: 0;
  }
`;
