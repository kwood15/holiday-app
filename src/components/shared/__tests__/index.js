import React from 'react';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
import { renderWithTheme } from '../helpers';
import { GlobalStyles } from '../GlobalStyles';
import { UtilitiesStyles } from '../UtilitiesStyles';
import {
  ContainerWrapper,
  HeaderWrapper,
  FooterWrapper,
  NavigationLink,
  BreadcrumbLink,
  Button,
  Input,
  BaseLink
} from '../SharedStyles';

describe('Styled components', () => {
  it('renders the correct <GlobalStyles /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<GlobalStyles />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <UtilitiesStyles /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<UtilitiesStyles />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <Button /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <ContainerWrapper /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<ContainerWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <HeaderWrapper /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<HeaderWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <BaseLink /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(
      <MemoryRouter>
        <BaseLink to="/test" />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <NavigationLink /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(
      <MemoryRouter>
        <NavigationLink to="/test" />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <BreadcrumbLink /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(
      <MemoryRouter>
        <BreadcrumbLink to="/test" />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <Input /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct <FooterWrapper /> styles based on the {theme} prop values', () => {
    const tree = renderWithTheme(<FooterWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
