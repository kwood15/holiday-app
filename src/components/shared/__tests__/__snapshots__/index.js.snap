// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled components renders the correct <BaseLink /> styles based on the {theme} prop values 1`] = `
.c0 {
  color: #0ea7b5;
  border: 2px solid #0ea7b5;
  background-color: #ffffff;
  padding: 0.6rem 1rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 1rem;
  font-weight: 700;
}

.c0:hover,
.c0:focus {
  color: #0c457d;
  border-color: #0c457d;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c0:disabled:hover,
.c0:disabled:focus {
  color: #505050;
  border-color: #505050;
}

<a
  className="c0"
  href="/test"
  onClick={[Function]}
/>
`;

exports[`Styled components renders the correct <BreadcrumbLink /> styles based on the {theme} prop values 1`] = `
.c0 {
  color: #505050;
}

.c0.active {
  color: #0ea7b5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  aria-current={null}
  className="c0"
  href="/test"
  onClick={[Function]}
/>
`;

exports[`Styled components renders the correct <Button /> styles based on the {theme} prop values 1`] = `
.c0 {
  color: #0ea7b5;
  border: 2px solid #0ea7b5;
  background-color: #ffffff;
  padding: 0.6rem 1rem;
  font-size: 1rem;
  cursor: pointer;
}

.c0:hover,
.c0:focus {
  color: #0c457d;
  border-color: #0c457d;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c0:disabled:hover,
.c0:disabled:focus {
  color: #505050;
  border-color: #505050;
}

.c0.modal__close {
  position: absolute;
  right: 0;
  top: 0;
  border: 0;
}

<button
  className="c0"
/>
`;

exports[`Styled components renders the correct <ContainerWrapper /> styles based on the {theme} prop values 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  max-width: 1200px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
/>
`;

exports[`Styled components renders the correct <FooterWrapper /> styles based on the {theme} prop values 1`] = `
.c0 {
  margin-top: 32px;
  background-color: #6bd2db;
  box-shadow: 0 0 16px rgba(0,0,0,0.3);
}

<footer
  className="c0"
/>
`;

exports[`Styled components renders the correct <GlobalStyles /> styles based on the {theme} prop values 1`] = `null`;

exports[`Styled components renders the correct <HeaderWrapper /> styles based on the {theme} prop values 1`] = `
.c0 {
  background: #ffffff;
  padding: 20px;
  box-shadow: 0 0 16px rgba(0,0,0,0.3);
}

<header
  className="c0"
/>
`;

exports[`Styled components renders the correct <Input /> styles based on the {theme} prop values 1`] = `
.c0 {
  font-size: 1rem;
  border: 2px solid #e0e0e0;
  padding: 0.6rem;
}

<input
  className="c0"
/>
`;

exports[`Styled components renders the correct <NavigationLink /> styles based on the {theme} prop values 1`] = `
.c0 {
  color: #505050;
}

.c0:hover,
.c0:focus {
  color: #0ea7b5;
  border-bottom: 2px solid #ffbe4f;
}

<a
  aria-current={null}
  className="c0"
  href="/test"
  onClick={[Function]}
/>
`;

exports[`Styled components renders the correct <UtilitiesStyles /> styles based on the {theme} prop values 1`] = `null`;
